### PR TITLE
Add sortable news and testing news list for exercise 2

### PR DESCRIPTION
**Pull request by RespectableRuessel**
Pull request for exercise 2.
Contains:
* Sort news asc / desc
* Show message when no news available
* Tests for news list
* Github CI

### CODE REVIEW

File: webapp/src/components/NewsForm/NewsForm.stories.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      // optional selector which element to inspect
+      element: '#root',
+      // axe-core configurationOptions (https://github.com/dequelabs/axe-core/blob/develop/doc/API.md#parameters-1)
+      config: {},
+      // axe-core optionsParameter (https://github.com/dequelabs/axe-core/blob/develop/doc/API.md#options-parameter)
+      options: {},
+      // optional flag to prevent the automatic check
+      manual: true,
+    },
+  }, */
```
Comment (Marcelloloco) [inline webapp/src/components/NewsForm/NewsForm.stories.ts:22]: Here is some code commented out, better remove it if not needed before merging to master

File: webapp/src/main.ts [hunk 2]
```diff
@@ -1,8 +1,29 @@
 import Vue from 'vue';
 import App from './App.vue';
+// const path = require('path');
```
Comment (Marcelloloco) [inline webapp/src/main.ts:3]: Could this also be some forgotten variable?

**Review by Marcelloloco — COMMENTED** [review]
Looks all good. Just check if that commented out code is still used to keep the main clean.

**Comment by roschaefer** [comment]
@Marcelloloco send a clear signal if this PR can be merged or must be improved.

Screenshot - 2020-11-09T182752 755


> :star: Review a pull request of another team.
> - Either "Request Changes" or "Approve" *do not just "Comment"*.


See: https://github.com/Systems-Development-and-Frameworks/bug-free-disco/pull/5#issuecomment-724141739

File: webapp/src/main.ts (hunk 2 above)
Comment (roschaefer) [reply webapp/src/main.ts:3]: Screenshot - 2020-11-09T183144 040


@Marcelloloco use the suggestion button :point_up: 
```suggestion
```

**Review by roschaefer — CHANGES_REQUESTED** [review]
invisible

Your "Files changed" do not contain your tests and is missing a lot of your production code, making it impossible to suggest changes on your code.

Could you please send another PR, maybe point it into another target branch where those changes are not present, and request another review?

There are a lot of possible refactorings and code smells, also I see runtime errors.
```
main.ts?cd49:12 Uncaught TypeError: Cannot assign to read only property 'exports' of object '#<Object>'
    at Module.eval (main.ts?cd49:12)
    at eval (main.ts:44)
    at Module../src/main.ts (app.js:1223)
    at __webpack_require__ (app.js:849)
    at fn (app.js:151)
    at Object.1 (app.js:1236)
    at __webpack_require__ (app.js:849)
    at checkDeferredModules (app.js:46)
    at app.js:925
    at app.js:928
```

Screenshot - 2020-11-09T200649 577

E.g. here is a suggestion how to refactor a test:
```
 describe('NewsList.vue', () => {
-  test('renders list is empty message when empty', async () => {
-    const msg = 'list is empty :(';
-    const wrapper = mount(NewsList);
-    const items = wrapper.findAllComponents(NewsItem);
-
-    for (let i = 0; i < items.length; i++) {
-      const news = items.at(i).vm.$props.news;
-      await items.at(i).vm.$emit('remove', news);
-    }
-
-    expect(wrapper.html()).toContain(msg);
+  test('renders "list is empty" message when empty', async () => {
+    const wrapper = mount(NewsList, { propsData: { initialItems: [] } });
+    expect(wrapper.text()).toContain('list is empty :(');
   });

   test('renders <NewsItem> for each item', () => {
-    const wrapper = mount(NewsList);
-    const items = wrapper.vm.$data.newsItems;
-
-    expect(wrapper.findAllComponents(NewsItem).length).toBe(items.length);
+    const initialItems = [
+      {id: 1, title: 'First title', votes: 0},
+      {id: 2, title: 'Second title', votes: 0},
+    ];
+    const wrapper = mount(NewsList, { propsData: { initialItems } });
+    expect(wrapper.findAllComponents(NewsItem)).toHaveLength(2);
   });

   test('toggles between ascending and descending order', async () => {
     const wrapper = mount(NewsList);
-    const startingState = wrapper.vm.$data.currentSortingOrder;
-
-    wrapper.find('button').trigger('click');
-
-    expect(startingState).not.toBe(wrapper.vm.$data.currentSortingOrder);
+    const initialItems = [
+      {id: 1, title: 'First title', votes: 0},
+      {id: 2, title: 'Second title', votes: 1},
+    ];
+    const wrapper = mount(NewsList, { propsData: { initialItems } });
+    expect(wrapper.findAllComponent(NewsItem).wrappers.map((w) => w.text()))
+      .toEqual(['Second title (1)', 'First title (0)']);
+
+    await wrapper.find('button').trigger('click');
+
+    expect(wrapper.findAllComponent(NewsItem).wrappers.map((w) => w.text()))
+      .toEqual(['First title (0)', 'Second title (1)']);
   });
```

File: webapp/src/components/NewsForm/NewsForm.stories.ts (hunk 1 above)
Comment (Marcelloloco) [inline webapp/src/components/NewsForm/NewsForm.stories.ts:22]:
```suggestion

```

**Review by Marcelloloco — CHANGES_REQUESTED** [review]
If that comment isn't used anymore then you should remove it.

PS: I think I accidentally committed sth where @roschaefer replied to my comment. Sorry for that

File: webapp/src/components/NewsList/NewsList.vue [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  }
+
+  public sortAsc(): void {
+    this.newsItems = this.newsItems.sort((a, b) => a.votes - b.votes);
+    this.currentSortingOrder = SortingOrder.Asc;
+  }
+
+  private sortNews(order: SortingOrder = SortingOrder.Desc): void {
+  switch (order) {
+      case SortingOrder.Desc:
```
Comment (roschaefer) [inline webapp/src/components/NewsList/NewsList.vue:99]: You have only 2 options right? A guard-clause would be enough:
```js
if (order === SortingOrder.Desc) return this.sortDesc()
return this.sortAsc()
```

**Review by roschaefer — APPROVED** [review]
well done

Here's what I found:
:star: Setup linter
:star: Setup jest
:star: Feature: Toggle ascending/descending order
:star: Feature: Empty state
:star: Co-locate stories, tests and component code
:star: Setup build server
:star: Setup storybook
:star: Storybook interactively controllable
:star: Storybook passes a11y tests
:star: Give a code review

You achieved 10/10 :star: very good!

Could you link your code reviews in your PR description, next time?
https://github.com/Systems-Development-and-Frameworks/bug-free-disco/pull/5#pullrequestreview-528232761
https://github.com/Systems-Development-and-Frameworks/bug-free-disco/pull/5#pullrequestreview-528299098

See my further suggestions below :point_down:

**Pull request merged**